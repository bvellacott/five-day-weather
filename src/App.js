import React, { Component } from 'react';
import './App.css';
import LocationChooser from './components/LocationChooser'
import WeatherDays from './components/WeatherDays';

class App extends Component {
	constructor(props) {
		super(props);
    this.db = props.db;

    this.state = {
      location: '',
      dayLists: []
    }
    this.db.getDayLists().then((lists) => { 
      this.setState({ location: this.db.data.city.name, dayLists: lists }); 
    });

    this.setLocation = this.setLocation.bind(this);
	}

  setLocation(newLoc) {
    this.db.setLocation(newLoc);
    this.setState({ location: 'Please wait...', dayLists: [] })
    this.db.getDayLists().then(lists => {
      this.setState({ location: newLoc, dayLists: lists })
    }).catch(err => { throw new Error(err); });
  }

  render() {
    return (
      <ul className="App">
      	<LocationChooser options={this.db.getLocations()} location={this.state.location} setLocation={this.setLocation}/>
        <WeatherDays lists={this.state.dayLists}/>
      </ul>
    );
  }
}

export default App;
