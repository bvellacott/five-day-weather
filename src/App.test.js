import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import createDbWithMocks from './lib/createDbWithMocks.test.js';

test('App scenario', async () => {
	var db = createDbWithMocks({
		location1: {
			city: { name: 'location1' },
			list: [
		    [{
		      "dt": 1479438000,
		      "weather": [{
		        "description": "light rain",
		        "icon": "10n"
		      }],
		    }],
		  ]
		},
		location2: {
			city: { name: 'location2' },
			list: [
		    [{
		      "dt": 1479438000,
		      "weather": [{
		        "description": "light rain",
		        "icon": "10n"
		      }],
		    }, {
		      "dt": 1479448800,
		      "weather": [{
		        "description": "clear sky",
		        "icon": "01n"
		      }],
		    }],
		    [{
		      "dt": 1479438123,
		      "weather": [{
		        "description": "heavy rain",
		        "icon": "13n"
		      }],
		    }]
		  ]
		},
	});
  const component = renderer.create( <App db={db}/> );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

	db.setLocation('location1');
	await db.getData();

  tree.children[0].children[0].props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

// Below code crashes the test, I assume because the app reloads the dom after a promise resolves 
  // tree.children[0].children[1].children[1].children[0].props.onClick();
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});
