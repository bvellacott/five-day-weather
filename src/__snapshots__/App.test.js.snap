exports[`test App scenario 1`] = `
<ul
  className="App">
  <div
    className="LocationChooser">
    <h1
      onClick={[Function]}>
      
    </h1>
  </div>
  <ul
    className="App" />
</ul>
`;

exports[`test App scenario 2`] = `
<ul
  className="App">
  <div
    className="LocationChooser">
    <h1
      onClick={[Function]}>
      
    </h1>
    <ul>
      <li>
        <span
          onClick={[Function]}>
          location1
        </span>
      </li>
      <li>
        <span
          onClick={[Function]}>
          location2
        </span>
      </li>
    </ul>
  </div>
  <ul
    className="App" />
</ul>
`;
